cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -60,14 +60,13 @@ func (s *server) Start(ctx context.Context, host component.Host) error {
 	if err != nil {
 		return fmt.Errorf("cannot find primary storage %s: %w", s.config.Storage.TracesPrimary, err)
 	}
+	f = storagemetrics.NewDecoratorFactory(f, baseFactory)
```

Comment:
Why do you apply the decorator in `jaegerquery` extension? I would do it in the storage extension, and attach a label `storage_kind={name}`